simulator/gdb/sim_wrapper.cpp

```diff
@@ -0,0 +1,78 @@
+/**
+ * sim_wrapper.cpp - Functional simulator wrappers for GDB
+ * @author Vyacheslav Kompan
+ * Copyright 2018 MIPT-MIPS
+ */
+
+#include "sim_wrapper.h"
+
+/* MIPT-MIPS simulator interfaces */
+#include <simulator.h>
+#include <infra/config/config.h>
+#include <mips/mips.h>
+#include <func_sim/func_sim.h>
+
+#define ISA_NAME "mips32"
+
+typedef struct {
+    std::unique_ptr<Simulator> ptr = nullptr;
+    std::string filename;
+} SimulatorInstance;
+static SimulatorInstance simInstance;
+
+
+/* Create simulator */
+void mips_sim_create (const char **argv, const char *filename) try {
```

Comment:
Has to be `const char* argv[]` or `const char* const * argv`

Reply:
The problem is that `config::handleArgs` accepts `const char **`, and it can't be cast from `const char* const *` or `char* const *` with `static_cast`. So I have to cast it explicitly in C code, removing const-qualifier.

Reply:
> The problem is that `config::handleArgs` accepts `const char **`

No it is not.

```c++
void handleArgs( int argc, const char* argv[]);
```

Reply:
Ok, I played with Godbolt's a little bit, it seems that `handleArgs` signature has to be:
```c++
void handleArgs( int argc, const char* const argv[]);
```

I'll push the change in a minute